documentation/guide/writing-persistent-objects.rst

```diff
@@ -0,0 +1,533 @@
+   False
+   >>> book.add_author('Carlos')
+   >>> bool(book._p_changed)
+   False
+   >>> bool(book.authors._p_changed)
+   True
+   >>> db.close()
+
+Note that in this example, when we added an author, the book itself
+didn't change, but the ``authors`` attribute value did.  Because
+``authors`` is a persistent object, it's stored in a separate database
+record from the book record and is managed by ZODB independent of the
+management of the book.
+
+In addition to ``PersistentList`` and ``PersistentMapping``, general
+persistent data structures are provided by the ``BTrees`` package,
+most notably ``BTree`` and ``TreeSet`` objects.  Unlike
+``PersistentList`` and ``PersistentMapping``, ``BTree`` and
+``TreeSet`` objects are scalable and can easily hold millions of
+objects, because their data are spread over many subobjects.
+
+It's generally better to use ``BTree`` objects than
+``PersistentMapping`` objects, because they're scalable and because
+they handle :ref:`conflicts <conflicts-label>` better. ``TreeSet``
+objects are the only ZODB-provided persistent set implementation.
+``BTree`` and ``TreeSets`` come in a number of families provided via
+different modules and differ in their internal implementations:
+
+===============  ===============  ================
+Module           Key type         Value Type
+===============  ===============  ================
+BTrees.OOBTree   object           object
+BTrees.IOBTree   integer          Object
+BTrees.OIBTree   object           integer
+BTrees.IIBTree   integer          integer
+BTrees.IFBTree   integer          float
+BTrees.LOBTree   64-bit integer   Object
+BTrees.OLBTree   object           64-bit integer
+BTrees.LLBTree   64-bit integer   64-bit integer
+BTrees.LFBTree   64-bit integer   float
+===============  ===============  ================
+
+Here's a version of the example that uses a ``TreeSet``::
+
+  import persistent
+  from BTrees.OOBTree import TreeSet
+
+  class Book(persistent.Persistent):
+
+     def __init__(self, title):
+         self.title = title
+         self.authors = TreeSet()
+
+     def add_author(self, author):
+         self.authors.add(author)
+
+.. -> src
+
+   >>> exec(src)
+   >>> db = ZODB.DB(None)
+   >>> with db.transaction() as conn:
+   ...     conn.root.book = Book("ZODB")
+   >>> conn = db.open()
+   >>> book = conn.root.book
+   >>> bool(book._p_changed)
+   False
+   >>> book.add_author('Carlos')
+   >>> bool(book._p_changed)
+   False
+   >>> bool(book.authors._p_changed)
+   True
+   >>> db.close()
+
+Properties
+==========
+
+If you implement some attributes using Python properties (or other
+types of descriptors), they are treated just like any other attributes
+by the persistence machinery.  When you set an attribute through a
+property, the object is considered changed, even if the property
+didn't actually modify the object state.
+
+Special attributes
+==================
+
+There are some attributes that are treated specially.
```

Comment:
Weird, I wonder why I didn't see that.

The problem I have with those docs is that they're too low level and deal too much with internal aspects (e.g. stickiness).

I have no problem with linking to them though.